### PR TITLE
Run from anywhere, build for different ARCH (Third time)

### PR DESCRIPTION
**Pull request by zorglube**
First sorry for the big PR.

Update the build part of the man, and add section about FreeBSD deployement.

Add the posiblity to run MN from anywhere in the filsystem : ./MovieNight or ./dir/dir/dir/MovieNight or c:/dir/.../dir/MovieNight

Add configuration file as argument : ./MovieNight -f ./dir/dir/.../settings.json

Modifie the Makefile to make some corss-compilation, to build into Windows, Linux, FreeBSD.

Any comment are welcome !

### CODE REVIEW

File: common/chatdata.go [hunk 1]
```diff
@@ -89,38 +91,51 @@ type DataMessage struct {
 	Type    MessageType
 }
```
Comment (zorchenhimer) [inline common/chatdata.go:93]: All the changes to this file should be in a separate pull request.

File: readme.md [hunk 2]
```diff
@@ -10,40 +9,43 @@
...
 ## Build requirements
-
 - Go 1.13 or newer
 - GNU Make
 
 ### Older Go Versions
-
 You can install a newer version of Go alongside your OS's distribution by
-following the guide here: [https://golang.org/doc/manage-install](https://golang.org/doc/manage-install)
+following the guide here: [https://golang.org/doc/install#extra_versions](https://golang.org/doc/install#extra_versions)
```
Comment (zorchenhimer) [inline readme.md:30]: This change should be reverted.  The correct URL is `https://golang.org/doc/manage-install`.

File: readme.md [hunk 3]
```diff
@@ -10,40 +9,43 @@
...
 Once you have that setup add an enviromnent variable named `GO_VERSION` and
 set it to the version you installed (eg, `1.14.1`).  The Makefile will now use
 the newer version.
 
 ### Compile and install
-
-To just download and run:
-
+You have to : 
+- download **git clone https://github.com/zorchenhimer/MovieNight**, go into the source directory **cd MovieNight**;
```
Comment (zorchenhimer) [inline readme.md:38]: Use inline code blocks for the commands.  ie, 
```
`git clone https://github.com/zorchenhimer/MovieNight`
```
instead of
```
**git clone https://github.com/zorchenhimer/MovieNight**
```

Same thing for the rest of the inline commands (the example block is fine as-is).

File: readme.md [hunk 4]
```diff
@@ -84,8 +86,10 @@ This docker-compose file will create a volume called *movienight-config* and aut
 #### Notes for Running Using docker-compose
 The container needs to be restarted to apply any changes you make to *settings.json*.
 
-## Usage
+### FreeNAS - FreeBSD build and run
+An script wich setup an Jail and build and run MovieNight into that Jail as been writen, you'll find it here [freenas-iocage-movienight] (https://github.com/zorglube/freenas-iocage-movienight)
```
Comment (zorchenhimer) [inline readme.md:90]: Remove the space between `]` and `(` so the link renders correctly.

**Review by zorchenhimer — CHANGES_REQUESTED** [review]

File: common/chatdata.go (hunk 1 above)
Comment (zorglube) [reply common/chatdata.go]: Allmost

Comment (zorglube) [reply common/chatdata.go]: Done, the Other PR is on it's way.

Comment (zorglube) [reply common/chatdata.go]: https://github.com/zorchenhimer/MovieNight/pull/143

File: readme.md (hunk 2 above)
Comment (zorglube) [reply readme.md]: Done

File: readme.md (hunk 3 above)
Comment (zorglube) [reply readme.md]: Done

File: readme.md (hunk 4 above)
Comment (zorglube) [reply readme.md]: Done
